Basecoin docs

davekaj opened the pull request:
Hey guys, I put together a document explaining how to get basecoin up and running. Patched it together from a few places that had out of date documents on basecoin. See below for reference:

**Basecoin explained:** https://github.com/cosmos/cosmos-sdk/blob/develop/docs/sdk/apps.md
**old basecoin docs :** https://github.com/cosmos/cosmos-sdk/blob/develop/docs/old/basecoin/basics.rst
**staking docs which were helpful:** https://github.com/cosmos/cosmos-sdk/blob/develop/docs/staking/intro.rst
**current testnet docs were also helpful:** https://github.com/cosmos/testnets/blob/master/README.md

As per #1027 there were no instructions on how to launch basecoin, so I looked through old docs until I figured out how to get basecoin to work with the current master branch.  Now this document should allow anyone to set it up.

Let me know if there is anything else that should be added and I can do it (i.e. democoin, more explanation on basecoin, more automatic commands instead of getting the user to copy paste long addresses). If not I will do one more run through of testing it and fixing any small errors in commands or spelling, and then ping that it is ready to be merged

zramsay commented:
@davekaj thanks! This is great. I'm going to do a run-through today - is there anything you wanted to add? We'll have to convert `.md` to `.rst` (see more about our docs [here](https://github.com/tendermint/coding#documentation) and a link to a tool that does the convertion to rst automatically) and move it to `docs/sdk/examples.rst` then add that line to `docs/index.rst` so it can get included on docs website.

davekaj commented:
Hey @zramsay , so i added in the more in depth details on basecoin from the old docs, as I think they add some good details for people new to the project. I updated the accounts section to reflect some changes in the types in the sdk. The transaction section talks about fees and gas, however that isn't implemented in basecoin yet.

But I mention that. I think it is ok to have the full description of how tx's will work, but this section could be removed if wanted. 

I did some small updates after going over it today and trying to think from a newbies perspective. 

So I think the content is good now! 

Sounds good about moving the doc and using .rst. Let me know if you need any more help setting up a doc like this for democoin or a similar example. I'm playing around with the examples quite a bit on my own, so it would be easy for me to help here. If i help again I'll be sure to use .rst!

davekaj commented:
hmm I'll also suggest removing the old basecoin doc here https://github.com/cosmos/cosmos-sdk/blob/develop/docs/old/basecoin/basics.rst 

as a developer from the outside I always try to check all documentation , and in this case I did have to go to the old documentation to find my way through setting up basecoin, but I think with this added in content, everything from the old doc can be removed.

zramsay commented on examples/examples.md:
```diff
@@ -1,6 +1,6 @@
 # Basecoin Example
 
-Here we explain how to get started with a basic Basecoin blockchain, how
+Here we explain how to get started with a simple Basecoin blockchain, how
```
i try to avoid using the word "simple" in docs. read a study once upon a time that showed noobs get turned off from a project really easily if it's described as simple and they stumble at all. basic is fine :)

zramsay commented on examples/examples.md:
````diff
@@ -38,7 +38,7 @@ basecli version
 ```
 
-They should read something like `0.17.1-5d18d5f`, but the versions will be constantly updating so don't worry if your version is higher that 0.17.1.
+They should read something like `0.17.1-5d18d5f`, but the versions will be constantly updating so don't worry if your version is higher that 0.17.1. That's a good thing.
````
haha :)

zramsay commented on examples/examples.md:
```diff
@@ -105,7 +105,7 @@ You should now see alice, bob and charlie's account all show up.
 NAME:	ADDRESS:					                PUBKEY:
 alice   90B0B9BE0914ECEE0B6DB74E67B07A00056B9BBD	1624DE62201D47E63694448665F5D0217EA8458177728C91C373047A42BD3C0FB78BD0BFA7
-bob   29D721F054537C91F618A0FDBF770DA51EF8C48D	1624DE6220F54B2A2CA9EB4EE30DE23A73D15902E087C09CC5616456DDDD3814769E2E0A16
+bob     29D721F054537C91F618A0FDBF770DA51EF8C48D	1624DE6220F54B2A2CA9EB4EE30DE23A73D15902E087C09CC5616456DDDD3814769E2E0A16
```
whitespace?

zramsay commented on examples/examples.md:
````diff
@@ -151,25 +151,28 @@ Note how we use the ``--name`` flag to select a different account to send from.
 ```
 
-Notice that the sequence is now 1, since we have already recorded bobs 1st transaction as sequnce 0. Also note the ``hash`` value in the response  in the terminal - this is the hash of the transaction. We can query for the transaction with this command:
+Notice that the sequence is now 1, since we have already recorded bobs 1st transaction as `sequnce 0`. Also note the ``hash`` value in the response  in the terminal - this is the hash of the transaction. We can query for the transaction with this command:
````
typo: sequnce -> sequence

davekaj commented:
@zramsay comments have been addressed!

ValarDragon commented on examples/examples.md:
```diff
@@ -0,0 +1,321 @@
+- `name` is the name you gave your key
+- `mycoin` is the name of the token for this basecoin demo, initialized in the genesis.json file
+- `sequence` is a tally of how many transactions have been made by this account. Since this is the first tx on this account, it is 0
+- `chain-id` is the unique ID that helps tendermint identify which network to connect to. You can find it in the terminal output from the gaiad daemon in the header block , or in the genesis.json file  at `~/.basecoind/config/gensis.json`
```
minor typo, `/gensis.json` should be `/genesis.json`

rigelrozanski commented on examples/examples.md:
````diff
@@ -0,0 +1,321 @@
+basecli tx <INSERT HASH HERE>
+```
+
+It will return the details of the transaction hash, such as how many coins were send and to which address, and on what block it occured.
````
occurred

davekaj replied:
not sure I follow, what occurred? did details of the tx hash not actually get returned with this info?

ValarDragon replied:
He was pointing out a typo. The word occurred was mispelled. In the last line you wrote `occured` with 1 `r`.

davekaj replied:
hahahah yes that makes perfect sense. Thank you!

rigelrozanski commented:
If you felt like it - I think the place this document should ultimately be going is around greater discussions as to how to construct your own application based on basecoin

davekaj commented:
@rigelrozanski i think thats a good point. No one wants to use the sdk to just build basecoin. they want to build their own apps off of basecoin. So good docs would look more like - 1) set up basecoin. show some functionality of basecoin 2) show how to add a module. - show that modules functionality. 3) explain how to add your own modules. etc. 

I could help with this in the future as I will definently be looking into this myself. If its decided to do the docs  this  way just open an issue and tag me in it

davekaj commented:
typos are fixed
